src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -126,7 +129,38 @@ export const portfolioApi = createApi({
           if (!adapter) throw new Error(`no adapter for ${chainId} not available`)
           const portfolioAccounts = { [pubkey]: await adapter.getAccount(pubkey) }
           const nftCollectionsById = selectNftCollections(state)
-          const data = accountToPortfolio({ portfolioAccounts, assetIds, nftCollectionsById })
+
+          const data = ((): Portfolio => {
+            switch (chainId) {
+              // add placeholder assets (evm tokens) for applicable chains where there is substantial missing data from coingecko
+              case arbitrumNovaChainId: {
+                const account = portfolioAccounts[pubkey] as Account<EvmChainId>
+
+                const assets = (account.chainSpecific.tokens ?? []).reduce<AssetsState>(
+                  (prev, token) => {
+                    // ignore existing assets and nfts (handled elsewhere)
+                    if (state.assets.byId[token.assetId] || isNft(token.assetId)) return prev
+                    prev.byId[token.assetId] = makeAsset({ ...token })
+                    prev.ids.push(token.assetId)
+                    return prev
+                  },
+                  { byId: {}, ids: [] },
+                )
+
+                // upsert placeholder assets
+                dispatch(assetSlice.actions.upsertAssets(assets))
+
+                return accountToPortfolio({
+                  portfolioAccounts,
+                  assetIds: assetIds.concat(assets.ids),
```

Comment:
chose this as it seemed more efficient than awaiting the `upsertAssets` and using the state directly